applications/asset_tracker/src/main.c

```diff
@@ -1112,18 +1112,21 @@ void main(void)
 
 		if ((fds[0].revents & POLLNVAL) == POLLNVAL) {
 			printk("Socket error: POLLNVAL\n");
+			printk("The cloud socket was unexpectedly closed.\n");
 			error_handler(ERROR_CLOUD, -EIO);
 			return;
 		}
 
 		if ((fds[0].revents & POLLHUP) == POLLHUP) {
 			printk("Socket error: POLLHUP\n");
+			printk("Connection was closed by the cloud.\n");
```

Comment:
This is the most likely scenario, but cannot be completely sure without inspecting the TCP headers going back and forth, I think

Reply:
Looks like it's pretty certain that's the reason: https://github.com/NordicPlayground/nrfxlib/blob/814a5297daec7f9b75238442fd69e569f693213e/bsdlib/include/nrf_socket.h#L1093